modules/aws/src/smithy4s/aws/AwsCredentialsFile.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.aws
+
+import cats.effect.Resource
+import cats.effect.Sync
+import cats.syntax.all._
+import java.nio.file.Paths
+import smithy4s.aws.kernel.AWS_ACCESS_KEY_ID
+import smithy4s.aws.kernel.AWS_SECRET_ACCESS_KEY
+import smithy4s.aws.kernel.AWS_SESSION_TOKEN
+import smithy4s.aws.kernel.AwsCredentials
+
+final case class AwsCredentialsFile(
+    default: Option[AwsCredentials],
+    profiles: Map[String, AwsCredentials]
+)
+
+object AwsCredentialsFile {
+  def loadFromDisk[F[_]](home: String, profile: Option[String])(implicit
+      F: Sync[F]
+  ): F[AwsCredentials] = {
+    loadFile(home)
+      .use(lines => processFileLines(lines).pure[F])
+      .flatMap(creds =>
+        profile
+          .map(p =>
+            creds.profiles
+              .get(p)
```

Comment:
you should be able to call `getOrElse(default)`  there

Reply:
IMO, if you ask for profile X and it's not found, it should fail

not fallback to the default without you knowing

Reply:
just let me know what you prefer and I'll get to it

but as an example, the AWS sdk does the following if you have a `aws_profile` env var, but not a corresponding profile in the credentials file:
```
ProfileCredentialsProvider(profileName=MY_PROFILE_SSO, profileFile=ProfileFile(profilesAndSectionsMap=[{default=Profile(name=default, properties=[region])}, {}])): Profile file contained no credentials for profile 'MY_PROFILE_SSO': ProfileFile(profilesAndSectionsMap=[{default=Profile(name=default, properties=[region])}, {}])
```

Reply:
> IMO, if you ask for profile X and it's not found, it should fail

Agreed.